doc/Hammer-Use/Hammer-Config.rst

```diff
@@ -0,0 +1,193 @@
+
+Basics
+------
+
+```
+foo:
+    bar:
+        adc: "yes"
+        dac: "no"
+```
+
+The basic idea of Hammer IR in YAML/JSON format is centered around a hierarchically nested tree of YAML/JSON dictionaries. For example, the above YAML snippet is translated to two variables which can be queried in code - `foo.bar.adc` would have `yes` and `foo.bar.dac` would have `no`.
+
+Overriding
+----------
+
+Hammer IR snippets frequently "override" each other. For example, a technology plugin might provide some defaults which a specific project can override with a project YAML snippet.
+
+For example, if the base snippet contains `foo: 12345` and the next snippet contains `foo: 54321`, then `get_setting("foo")` would return `54321`.
+
+Meta actions
+--------------
+
+Sometimes it is desirable that variables are not completely overwritten, but instead modified.
+
+For example, say that the technology plugin provides:
+
+```
+vlsi.tech.foobar65.bad_cells: ["NAND4X", "NOR4X"]
+```
+
+And let's say that in our particular project, we find it undesirable to use the `NAND2X` and `NOR2X` cells. However, if we simply put the following in our project YAML, the references to NAND4X and NOR4X disappear and we don't want to have to copy the information from the base plugin, which may change, or which may be proprietary, etc.
+
+```
+vlsi.tech.foobar65.bad_cells: ["NAND8X", "NOR8X"]
+```
+
+The solution is **meta variables**. This lets `hammer-config` know that instead of simply replacing the base variable, it should do a particular special action.
+
+In this case, we can use the `append` meta action:
+
+```
+vlsi.tech.foobar65.bad_cells: ["NAND8X", "NOR8X"]
+vlsi.tech.foobar65.bad_cells_meta: append
+```
+
+This will yield the desired result of `["NAND4X", "NOR4X", "NAND8X", "NOR8X"]` when `get_settings("vlsi.tech.foobar65.bad_cells")` is called in the end.
+
+Applying multiple meta actions
+------------------------------
+
+Multiple meta actions can be applied sequentially if the `_meta` variable is an array. Example:
+
+Layer 1:
+```
+foo.flash: yes
+```
+
+Layer 2 (located at /opt/foo):
+```
+foo.pipeline: "CELL_${foo.flash}.lef"
+foo.pipeline_meta: ['subst', 'prependlocal']
+```
+
+
+Result: `get_setting("foo.pipeline")` = `/opt/foo/CELL_yes.lef`
+
+Common meta actions
+-------------------
+
+* `append`: append the elements provided to the base list
+* `subst`: substitute variables into a string.
+
+Base:
+```
+foo.flash: yes
+```
+
+Meta:
+```
+foo.pipeline: "${foo.flash}man"
+foo.pipeline_meta: subst
+```
+
+Result: `get_setting("foo.flash")` = `yesman`
+
+* `lazysubst`: by default, variables are only substituted from previous configs. Using `lazysubst` allows us to deter the substitution until the very end.
```

Comment:
I think a general motivation of the `lazy...` modifier (waiting til all overriding is reconciled) is warranted in the section about key overriding or introducing meta variables?

Reply:
Rescinding this after consideration.